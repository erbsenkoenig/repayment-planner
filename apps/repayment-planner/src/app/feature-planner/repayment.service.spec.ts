import { TestBed } from '@angular/core/testing';

import { RepaymentService } from './repayment.service';

describe('RepaymentService', () => {
  let service: RepaymentService;

  beforeEach(() => {
    TestBed.configureTestingModule({});
    service = TestBed.inject(RepaymentService);
  });

  it('should be created', () => {
    expect(service).toBeTruthy();
  });

  describe('calculateRepaymentPlan', () => {
    it('should handle loan ended before fixed interest rate is done', () => {
      expect(
        service.calculateRepaymentPlan({
          debitInterest: 2.12,
          loanAmount: 100,
          repaymentRate: 50,
          fixedInterestRate: 10,
        })
      ).toMatchSnapshot();
    });

    it('should handle loan with remaining loan after fixed interest rate', () => {
      expect(
        service.calculateRepaymentPlan({
          debitInterest: 2.12,
          loanAmount: 100,
          repaymentRate: 50,
          fixedInterestRate: 1,
        })
      ).toMatchSnapshot();
    });
  });
});
