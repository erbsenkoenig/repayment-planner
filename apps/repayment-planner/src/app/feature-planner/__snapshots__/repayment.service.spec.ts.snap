// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepaymentService calculateRepaymentPlan should handle loan ended before fixed interest rate is done 1`] = `
[
  {
    "id": 0,
    "interest": 0,
    "rate": -100,
    "repayment": -100,
    "residualDept": 100,
  },
  {
    "id": 1,
    "interest": 0.17666666666666667,
    "rate": 4.343333333333333,
    "repayment": 4.166666666666666,
    "residualDept": 95.83333333333333,
  },
  {
    "id": 2,
    "interest": 0.16930555555555554,
    "rate": 4.343333333333333,
    "repayment": 4.174027777777777,
    "residualDept": 91.65930555555555,
  },
  {
    "id": 3,
    "interest": 0.1619314398148148,
    "rate": 4.343333333333333,
    "repayment": 4.181401893518518,
    "residualDept": 87.47790366203704,
  },
  {
    "id": 4,
    "interest": 0.15454429646959877,
    "rate": 4.343333333333333,
    "repayment": 4.188789036863734,
    "residualDept": 83.2891146251733,
  },
  {
    "id": 5,
    "interest": 0.14714410250447282,
    "rate": 4.343333333333333,
    "repayment": 4.19618923082886,
    "residualDept": 79.09292539434443,
  },
  {
    "id": 6,
    "interest": 0.13973083486334184,
    "rate": 4.343333333333333,
    "repayment": 4.203602498469991,
    "residualDept": 74.88932289587444,
  },
  {
    "id": 7,
    "interest": 0.13230447044937818,
    "rate": 4.343333333333333,
    "repayment": 4.211028862883954,
    "residualDept": 70.6782940329905,
  },
  {
    "id": 8,
    "interest": 0.12486498612494988,
    "rate": 4.343333333333333,
    "repayment": 4.218468347208383,
    "residualDept": 66.4598256857821,
  },
  {
    "id": 9,
    "interest": 0.1174123587115484,
    "rate": 4.343333333333333,
    "repayment": 4.2259209746217845,
    "residualDept": 62.23390471116032,
  },
  {
    "id": 10,
    "interest": 0.10994656498971658,
    "rate": 4.343333333333333,
    "repayment": 4.233386768343617,
    "residualDept": 58.00051794281671,
  },
  {
    "id": 11,
    "interest": 0.1024675816989762,
    "rate": 4.343333333333333,
    "repayment": 4.240865751634357,
    "residualDept": 53.75965219118235,
  },
  {
    "id": 12,
    "interest": 0.09497538553775547,
    "rate": 4.343333333333333,
    "repayment": 4.2483579477955775,
    "residualDept": 49.51129424338677,
  },
  {
    "id": 13,
    "interest": 0.08746995316331664,
    "rate": 4.343333333333333,
    "repayment": 4.255863380170016,
    "residualDept": 45.25543086321676,
  },
  {
    "id": 14,
    "interest": 0.07995126119168294,
    "rate": 4.343333333333333,
    "repayment": 4.26338207214165,
    "residualDept": 40.992048791075106,
  },
  {
    "id": 15,
    "interest": 0.07241928619756603,
    "rate": 4.343333333333333,
    "repayment": 4.270914047135767,
    "residualDept": 36.72113474393934,
  },
  {
    "id": 16,
    "interest": 0.06487400471429283,
    "rate": 4.343333333333333,
    "repayment": 4.27845932861904,
    "residualDept": 32.442675415320295,
  },
  {
    "id": 17,
    "interest": 0.05731539323373252,
    "rate": 4.343333333333333,
    "repayment": 4.2860179400996,
    "residualDept": 28.156657475220694,
  },
  {
    "id": 18,
    "interest": 0.04974342820622323,
    "rate": 4.343333333333333,
    "repayment": 4.29358990512711,
    "residualDept": 23.863067570093584,
  },
  {
    "id": 19,
    "interest": 0.04215808604049867,
    "rate": 4.343333333333333,
    "repayment": 4.301175247292834,
    "residualDept": 19.56189232280075,
  },
  {
    "id": 20,
    "interest": 0.03455934310361466,
    "rate": 4.343333333333333,
    "repayment": 4.308773990229718,
    "residualDept": 15.253118332571034,
  },
  {
    "id": 21,
    "interest": 0.026947175720875495,
    "rate": 4.343333333333333,
    "repayment": 4.316386157612457,
    "residualDept": 10.936732174958578,
  },
  {
    "id": 22,
    "interest": 0.019321560175760154,
    "rate": 4.343333333333333,
    "repayment": 4.324011773157572,
    "residualDept": 6.612720401801006,
  },
  {
    "id": 23,
    "interest": 0.011682472709848445,
    "rate": 2.281069541177522,
    "repayment": 4.331650860623484,
    "residualDept": 2.281069541177522,
  },
]
`;

exports[`RepaymentService calculateRepaymentPlan should handle loan with remaining loan after fixed interest rate 1`] = `
[
  {
    "id": 0,
    "interest": 0,
    "rate": -100,
    "repayment": -100,
    "residualDept": 100,
  },
  {
    "id": 1,
    "interest": 0.17666666666666667,
    "rate": 4.343333333333333,
    "repayment": 4.166666666666666,
    "residualDept": 95.83333333333333,
  },
  {
    "id": 2,
    "interest": 0.16930555555555554,
    "rate": 4.343333333333333,
    "repayment": 4.174027777777777,
    "residualDept": 91.65930555555555,
  },
  {
    "id": 3,
    "interest": 0.1619314398148148,
    "rate": 4.343333333333333,
    "repayment": 4.181401893518518,
    "residualDept": 87.47790366203704,
  },
  {
    "id": 4,
    "interest": 0.15454429646959877,
    "rate": 4.343333333333333,
    "repayment": 4.188789036863734,
    "residualDept": 83.2891146251733,
  },
  {
    "id": 5,
    "interest": 0.14714410250447282,
    "rate": 4.343333333333333,
    "repayment": 4.19618923082886,
    "residualDept": 79.09292539434443,
  },
  {
    "id": 6,
    "interest": 0.13973083486334184,
    "rate": 4.343333333333333,
    "repayment": 4.203602498469991,
    "residualDept": 74.88932289587444,
  },
  {
    "id": 7,
    "interest": 0.13230447044937818,
    "rate": 4.343333333333333,
    "repayment": 4.211028862883954,
    "residualDept": 70.6782940329905,
  },
  {
    "id": 8,
    "interest": 0.12486498612494988,
    "rate": 4.343333333333333,
    "repayment": 4.218468347208383,
    "residualDept": 66.4598256857821,
  },
  {
    "id": 9,
    "interest": 0.1174123587115484,
    "rate": 4.343333333333333,
    "repayment": 4.2259209746217845,
    "residualDept": 62.23390471116032,
  },
  {
    "id": 10,
    "interest": 0.10994656498971658,
    "rate": 4.343333333333333,
    "repayment": 4.233386768343617,
    "residualDept": 58.00051794281671,
  },
  {
    "id": 11,
    "interest": 0.1024675816989762,
    "rate": 4.343333333333333,
    "repayment": 4.240865751634357,
    "residualDept": 53.75965219118235,
  },
  {
    "id": 12,
    "interest": 0.09497538553775547,
    "rate": 4.343333333333333,
    "repayment": 4.2483579477955775,
    "residualDept": 49.51129424338677,
  },
]
`;
